accord-core/src/test/java/accord/impl/basic/DelayedCommandStores.java

```diff
@@ -54,6 +54,20 @@ public static CommandStores.Factory factory(PendingQueue pending)
 
     public static class DelayedCommandStore extends InMemoryCommandStore
     {
+        private class DelayedTask<T> extends Task<T>
```

Comment:
in burn test I hit a case of something running in the wrong store... I don't see where that fix is in my patch, but pulled this forward to be correct...

if we only `unsafeRunIn` for the task, we fail to handle the callbacks... so need to run `super.run` in `unsafeRunIn` so callbacks also run in the same store by default (which matches the real world)...

Reply:
> The contract for what thread callbacks run in is kind of weak

Agree, it's just the same as `Future` where it has 2 possible outcomes: `Thread.current()` (the result is present), and the thread that set the future....  Now, if you know what thread will set the future then this isn't an issue... when it can be arbitrary or not "the correct one" you "should" call the methods with an `Executor` or call `withExecutor` to force the switch